api/envoy/api/v2/route/route.proto

```diff
@@ -926,7 +926,18 @@ message HeaderMatcher {
     // * For range [-10,0), route will match for header value -1, but not for 0, "somestring", 10.9,
     //   "-1somestring"
     envoy.type.Int64Range range_match = 6;
+
+    // If specified, header match will be be performed based on whether the header is in the request.
+    bool present_match = 7;
```

Comment:
The docs read as if the default behavior when `header_match_specifier` is missing is to act as if `present_match` is true, but really the behavior is to act as if `exact_match` is the empty string, which still makes using `invert_match` confusing: an empty `exact_match` and `invert_match` enabled will never match any request (by my reading).

I think we could fix this either by changing the default behavior -- no one can be using `invert_match` now and `present_match` should behave like empty `exact_match`. Then going forward `invert_match` works sensibly. (Changing the default after invert_match is in the wild could break someone.)

Reply:
This is tricky because it moves the `value` field from deprecated to broken.

`value` was deprecated in 1.6.0 and this would be for 1.7.0, so I *think* it's okay to break this, but would appreciate anyone confirming.

Reply:
I don't think we should break any existing behavior. Sorry, I don't think I fully grok the discussion. Is there a way to fix the concerns without breaking existing routing behavior?

Reply:
Late me try to restate my concern: the docs on the proto say the default behavior is to check for header presence when there is no `header_match_specifier` set (and really it also means when no `value` is set). That all works fine on master. After this change (setting aside the latest commits that change this behavior), it continues to work as documented unless you also set `invert_match` at which point no request will ever match: the HeaderMatchType will be Value, so the presence check is done and no header will result in a match failure (though if you'd set `present` = true it will work).

So one way to fix this is just make it clear that the default behavior is really `exact_match = ""` and that it will match any header value, but cannot be inverted. That seems a bit weird to me, but it would be fine.

I think the current changes can work with the docs unchanged if the default case only sets the HeaderMatchType to Present when "value" is the empty string.